src/buffer.c

```diff
@@ -325,7 +325,6 @@ extern int BUFFER_unbuild(BUFFER_HANDLE handle)
         BUFFER* b = (BUFFER*)handle;
         if (b->buffer != NULL)
         {
-            LogError("Failure buffer data is NULL");
```

Comment:
Instead of removing it completely, could you move it to the block below (that handles when it's actually null)?

Reply:
No, tried that already: see the original PR explanation. 
The buffer unbuild function is called for _any_ buffer - null or not. Since there is no way for the app to check if the buffer is null or not before calling (similar to `free(void*)`) we cannot error on either of these cases.

Reply:
(I've talked about this change offline with @jebrando - he may have some background on original design considerations. )

Reply:
Frankly that is a bad thing from uhttp. There is never a need to call "destroy"/"close"/"unbuild" with a NULL handle.

It can be convenience of design, but we can see how this convenience of design can lead to ignoring errors.
Ignoring errors is not good because it can lead to missing coding errors in other cases which leads to badly written code.

So I would suggest fixing this in the proper way (add the error when the handle is NULL).
And then uhttp has to also be fixed honestly.

---
In reply to: [339236793](https://github.com/Azure/azure-c-shared-utility/pull/405#discussion_r339236793) [](ancestors = 339236793)

Reply:
By "fixed in the right way" I mean move the log error between ln. 320 and 321.

---
In reply to: [339405110](https://github.com/Azure/azure-c-shared-utility/pull/405#discussion_r339405110) [](ancestors = 339405110,339236793)

Reply:
Like here:

BUFFER_HANDLE BUFFER_create(const unsigned char* source, size_t size)
{
    BUFFER* result;
    /*Codes_SRS_BUFFER_02_001: [If source is NULL then BUFFER_create shall return NULL.]*/
    if (source == NULL)
    {
        LogError("invalid parameter source: %p", source);
        result = NULL;
    }

---
In reply to: [339405202](https://github.com/Azure/azure-c-shared-utility/pull/405#discussion_r339405202) [](ancestors = 339405202,339405110,339236793)